libfreerdp/core/license.c

```diff
@@ -790,7 +823,7 @@ BOOL license_read_platform_challenge_packet(rdpLicense* license, wStream* s)
 	WLog_DBG(TAG, "MacData:");
 	winpr_HexDump(TAG, WLOG_DEBUG, MacData, 16);
 #endif
-	return TRUE;
+	return ret;
```

Comment:
if license_decrypt_platform_challenge fails some of the pointers printed in the debug lines above are null.
Suggest to do the debug statements inside a  if (ret)